### ISSUE TITLE
Mail::AddressList can not parse

### ISSUE THREAD

**Issue by jangrewe** [zammad/zammad#2572]
### Infos:

* Used Zammad version: stable / 2.9.x
* Installation method (source, package, ..): git
* Operating system: Debian stretch
* Database + version: MariaDB 10.2.24
* Elasticsearch version: 6.6
* Browser + version: Firefox 66

### Expected behavior:

* Mails get processed

### Actual behavior:

```
zammad@hades:~$ rails r 'Channel::EmailParser.process_unprocessable_mails'
Mail::AddressList can not parse |"Foo Šoobar" <fsoobar@gmail.com>|
Reason was: Only able to parse up to "Foo Š
"ERROR: Can't process email, you will find it for bug reporting under /opt/zammad/tmp/unprocessable_mail/d65d37fbec0d32fc5ce6a0b0cbf5e1e1.eml, please create an issue at https://github.com/zammad/zammad/issues"
"ERROR: #<ActiveRecord::StatementInvalid: Mysql2::Error: MySQL client is not connected: ROLLBACK>"
/opt/zammad/app/models/channel/email_parser.rb:129:in `rescue in process': #<ActiveRecord::StatementInvalid: Mysql2::Error: MySQL client is not connected: ROLLBACK> (RuntimeError)
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/mysql2-0.4.10/lib/mysql2/client.rb:120:in `_query'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/mysql2-0.4.10/lib/mysql2/client.rb:120:in `block in query'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/mysql2-0.4.10/lib/mysql2/client.rb:119:in `handle_interrupt'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/mysql2-0.4.10/lib/mysql2/client.rb:119:in `query'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/activerecord-5.1.6.2/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:214:in `block (2 levels) in execute'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/activesupport-5.1.6.2/lib/active_support/dependencies/interlock.rb:46:in `block in permit_concurrent_loads'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/activesupport-5.1.6.2/lib/active_support/concurrency/share_lock.rb:185:in `yield_shares'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/activesupport-5.1.6.2/lib/active_support/dependencies/interlock.rb:45:in `permit_concurrent_loads'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/activerecord-5.1.6.2/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:213:in `block in execute'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/activerecord-5.1.6.2/lib/active_record/connection_adapters/abstract_adapter.rb:613:in `block (2 levels) in log'
/opt/zammad/.rvm/rubies/ruby-2.4.4/lib/ruby/2.4.0/monitor.rb:214:in `mon_synchronize'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/activerecord-5.1.6.2/lib/active_record/connection_adapters/abstract_adapter.rb:612:in `block in log'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/activesupport-5.1.6.2/lib/active_support/notifications/instrumenter.rb:21:in `instrument'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/activerecord-5.1.6.2/lib/active_record/connection_adapters/abstract_adapter.rb:604:in `log'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/activerecord-5.1.6.2/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:212:in `execute'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/activerecord-5.1.6.2/lib/active_record/connection_adapters/mysql/database_statements.rb:26:in `execute'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/activerecord-5.1.6.2/lib/active_record/connection_adapters/abstract_mysql_adapter.rb:240:in `exec_rollback_db_transaction'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/activerecord-5.1.6.2/lib/active_record/connection_adapters/abstract/database_statements.rb:288:in `rollback_db_transaction'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/activerecord-5.1.6.2/lib/active_record/connection_adapters/abstract/query_cache.rb:17:in `rollback_db_transaction'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/activerecord-5.1.6.2/lib/active_record/connection_adapters/abstract/transaction.rb:135:in `rollback'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/activerecord-5.1.6.2/lib/active_record/connection_adapters/abstract/transaction.rb:185:in `block in rollback_transaction'
/opt/zammad/.rvm/rubies/ruby-2.4.4/lib/ruby/2.4.0/monitor.rb:214:in `mon_synchronize'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/activerecord-5.1.6.2/lib/active_record/connection_adapters/abstract/transaction.rb:183:in `rollback_transaction'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/activerecord-5.1.6.2/lib/active_record/connection_adapters/abstract/transaction.rb:197:in `rescue in block in within_new_transaction'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/activerecord-5.1.6.2/lib/active_record/connection_adapters/abstract/transaction.rb:214:in `block in within_new_transaction'
/opt/zammad/.rvm/rubies/ruby-2.4.4/lib/ruby/2.4.0/monitor.rb:214:in `mon_synchronize'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/activerecord-5.1.6.2/lib/active_record/connection_adapters/abstract/transaction.rb:191:in `within_new_transaction'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/activerecord-5.1.6.2/lib/active_record/connection_adapters/abstract/database_statements.rb:235:in `transaction'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/activerecord-5.1.6.2/lib/active_record/transactions.rb:210:in `transaction'
/opt/zammad/app/models/transaction.rb:10:in `execute'
/opt/zammad/app/models/channel/email_parser.rb:170:in `_process'
/opt/zammad/app/models/channel/email_parser.rb:111:in `block in process'
/opt/zammad/.rvm/rubies/ruby-2.4.4/lib/ruby/2.4.0/timeout.rb:93:in `block in timeout'
/opt/zammad/.rvm/rubies/ruby-2.4.4/lib/ruby/2.4.0/timeout.rb:33:in `block in catch'
/opt/zammad/.rvm/rubies/ruby-2.4.4/lib/ruby/2.4.0/timeout.rb:33:in `catch'
/opt/zammad/.rvm/rubies/ruby-2.4.4/lib/ruby/2.4.0/timeout.rb:33:in `catch'
/opt/zammad/.rvm/rubies/ruby-2.4.4/lib/ruby/2.4.0/timeout.rb:108:in `timeout'
/opt/zammad/app/models/channel/email_parser.rb:110:in `process'
/opt/zammad/app/models/channel/email_parser.rb:479:in `block in process_unprocessable_mails'
/opt/zammad/app/models/channel/email_parser.rb:478:in `glob'
/opt/zammad/app/models/channel/email_parser.rb:478:in `process_unprocessable_mails'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/railties-5.1.6.2/lib/rails/commands/runner/runner_command.rb:37:in `perform'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/railties-5.1.6.2/lib/rails/commands/runner/runner_command.rb:37:in `eval'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/railties-5.1.6.2/lib/rails/commands/runner/runner_command.rb:37:in `perform'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/railties-5.1.6.2/lib/rails/command/base.rb:63:in `perform'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/railties-5.1.6.2/lib/rails/command.rb:44:in `invoke'
/opt/zammad/.rvm/gems/ruby-2.4.4/gems/railties-5.1.6.2/lib/rails/commands.rb:16:in `<top (required)>'
bin/rails:9:in `require'
bin/rails:9:in `<main>'
        from /opt/zammad/app/models/channel/email_parser.rb:108:in `process'
        from /opt/zammad/app/models/channel/email_parser.rb:479:in `block in process_unprocessable_mails'
        from /opt/zammad/app/models/channel/email_parser.rb:478:in `glob'
        from /opt/zammad/app/models/channel/email_parser.rb:478:in `process_unprocessable_mails'
        from /opt/zammad/.rvm/gems/ruby-2.4.4/gems/railties-5.1.6.2/lib/rails/commands/runner/runner_command.rb:37:in `perform'
        from /opt/zammad/.rvm/gems/ruby-2.4.4/gems/railties-5.1.6.2/lib/rails/commands/runner/runner_command.rb:37:in `eval'
        from /opt/zammad/.rvm/gems/ruby-2.4.4/gems/railties-5.1.6.2/lib/rails/commands/runner/runner_command.rb:37:in `perform'
        from /opt/zammad/.rvm/gems/ruby-2.4.4/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
        from /opt/zammad/.rvm/gems/ruby-2.4.4/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
        from /opt/zammad/.rvm/gems/ruby-2.4.4/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
        from /opt/zammad/.rvm/gems/ruby-2.4.4/gems/railties-5.1.6.2/lib/rails/command/base.rb:63:in `perform'
        from /opt/zammad/.rvm/gems/ruby-2.4.4/gems/railties-5.1.6.2/lib/rails/command.rb:44:in `invoke'
        from /opt/zammad/.rvm/gems/ruby-2.4.4/gems/railties-5.1.6.2/lib/rails/commands.rb:16:in `<top (required)>'
        from bin/rails:9:in `require'
        from bin/rails:9:in `<main>'
```

### Steps to reproduce the behavior:

* Receive a mail with an unusual character in the From:
```
From: =?UTF-8?Q?Foo_=C5=A0oobar?= <fsoobar@gmail.com>
```

**Comment by SummerStorm** [comment]
Hi there, can you please anonymize the email, rename it as a `txt` file, and upload it here? Thanks!

**Comment by jangrewe** [comment]
Looks like this was 2 problems in one, as the "can not parse" error went away when i replaced that special character, but it still wouldn't import - until i increased `max_allowed_packet` for MariaDB.

The mail is now imported and gone, unfortunately, but the remaining issue seems to be characters like `Š` in the From.

**Comment by SummerStorm** [comment]
This is a duplicate of #2456, which has been fixed on the `develop` branch. The fix will be included in Zammad 2.10.

I tried this minimal example on `develop` and it imported the `From` field correctly:

```
From: =?UTF-8?Q?Foo_=C5=A0oobar?= <fsoobar@gmail.com>
To: <info@example.com>
Subject: test subject

body
```

Please re-open this issue if you experience it after Zammad 2.10 has been released. Thank you for your contribution.
